controllers/awscluster_controller.go

```diff
@@ -256,6 +267,14 @@ func (r *AWSClusterReconciler) reconcileDelete(ctx context.Context, clusterScope
 		allErrs = append(allErrs, errors.Wrap(err, "error deleting network"))
 	}
 
+	if err := iamService.DeleteOIDCProvider(ctx); err != nil {
```

Comment:
Does this need to be dependent on the feature flag?

Reply:
I added a feature flag gate here, and in the `ReconcileNormal`